modules/core/src/types/layer-props.ts

```diff
@@ -83,12 +83,15 @@ export type TextureSource =
  * * A `Promise` whose resolved value will be used as the value of the `data` prop.
  * * An `AsyncIterable` that yields data in batches. Each batch is expected to be an array of objects.
  * * `string` - a URL to load data from
+ * * `null` or `undefined` - empty data
  */
 export type LayerDataSource<DataType> =
   | LayerData<DataType>
   | string
   | AsyncIterable<DataType[]>
-  | Promise<LayerData<DataType>>;
+  | Promise<LayerData<DataType>>
+  | null
+  | undefined;
```

Comment:
Does undefined behave the same as null in layers? I've always favored null since I got convinced undefined didn't always behave the same way.. this was a long time ago though

Reply:
I guess I can drop `undefined`. The reason for adding this is to allow creating a layer in React while data is still loading.

Reply:
Oh, I wasn't familiar with that one. If it's been in use for some time than it's probably better to include it and wait for a bug report if it causes any issue